### PR TITLE
fix(avatar): don't use 'user/avatar' endpoint for generating a temp avatar

### LINKED ISSUE #13208: Set conversation avatar updates account avatar instead
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

Quite an edge case, but still happens under conditions described below.

## Steps to reproduce
1. Upload square picture via Files app (width = height)
2. Try to set conversation avatar "from Nextcloud"
3. See result

### Expected behaviour
Cropper should open (to show how picture would look like, and maybe crop some more)

### Actual behaviour

Frontend logic copied from server settings / AvatarSection:
https://github.com/nextcloud/spreed/blob/bd84203ddd2672fc017fa9883a315a0da2e20445/src/components/ConversationSettings/ConversationAvatarEditor.vue#L278-L288

If image is square, cropping isn't requested, and image is set by default endpoint => as avatar image
https://github.com/nextcloud/server/blob/af6de04e9e141466dc229e444ff3f146f4a34765/core/Controller/AvatarController.php#L243-L254

## Talk app

**Talk app version:** (main, but given server/frontend code is unchanged, also valid for 17)

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Fix #13208

### 🖼️ Screenshots / Screencasts

🏚️ Before | 🏡 After
-- | --
image | image

### 🏁 Checklist

- [ ] 🌏 Tested with different browsers / clients:
  - [x] Chromium (Chrome / Edge / Opera / Brave)
  - [ ] Firefox
  - [ ] Safari
  - [x] Talk Desktop
  - [ ] Not risky to browser differences / client
- [ ] ⛑️ Tests are included or not possible

### CODE REVIEW

File: src/components/ConversationSettings/ConversationAvatarEditor.vue [hunk 1]
```diff
@@ -268,24 +268,17 @@ export default {
...
-					this.$refs.cropper.replace(tempAvatar)
-					this.showCropper = true
-				} else {
-					showError(data.data.message)
-					this.cancel()
-				}
+				const tempAvatar = generateUrl(`/core/preview?fileId=${fileid}&x=512&y=512&a=1`)
+				this.loading = false
+				this.$refs.cropper.replace(tempAvatar)
+				this.showCropper = true
```
Comment (DorraJaouad) [inline src/components/ConversationSettings/ConversationAvatarEditor.vue:281]: Do we still need loading flags here?

**Review by DorraJaouad — APPROVED** [review]
Tested

File: src/components/ConversationSettings/ConversationAvatarEditor.vue (hunk 1 above)
Comment (Antreesy) [reply src/components/ConversationSettings/ConversationAvatarEditor.vue]: Looks like a no, we don't wait for server response here (even for preview load)

**Review by DorraJaouad — APPROVED** [review]
Tested

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**